site/profile/manifests/users.pp

```diff
@@ -144,16 +144,31 @@
 
   $public_keys.each | Integer $index, String $sshkey | {
     $split = split($sshkey, ' ')
-    if length($split) > 2 {
-      $keyname = String($split[2], '%t')
+    # support providing options:
+    if '=' in $split[0] {
```

Comment:
there is not necessarily a '=' in options. `restrict` is an option that does not have `=` in it